src/Analyzers/Core/Analyzers/SimplifyLinqExpression/AbstractSimplifyLinqExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,144 @@
+            {
+                return;
+            }
+
+            if (!TryGetLinqWhereExtensionMethods(enumerableType, out var whereMethodSymbols))
+            {
+                return;
+            }
+
+            if (!TryGetLinqMethodsThatDotNotReturnEnumerables(enumerableType, out var linqMethodSymbols))
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(
+                context => AnalyzeInvocationOperation(context, whereMethodSymbols, linqMethodSymbols),
+                OperationKind.Invocation);
+
+            static bool TryGetEnumerableTypeSymbol(Compilation compilation, [NotNullWhen(true)] out INamedTypeSymbol? enumerableType)
+            {
+                if (typeof(Enumerable)?.FullName is string fullyQualifiedName)
+                {
+                    enumerableType = compilation.GetTypeByMetadataName(fullyQualifiedName);
+                    return enumerableType is not null;
+                }
+
+                enumerableType = null;
+                return false;
+            }
+
+            static bool TryGetLinqWhereExtensionMethods(INamedTypeSymbol enumerableType, out ImmutableArray<IMethodSymbol> whereMethods)
+            {
+                whereMethods = enumerableType.GetMembers(nameof(Enumerable.Where)).OfType<IMethodSymbol>().ToImmutableArray();
+                return !whereMethods.IsEmpty;
+            }
+
+            static bool TryGetLinqMethodsThatDotNotReturnEnumerables(INamedTypeSymbol enumerableType, out ImmutableArray<IMethodSymbol> linqMethods)
+            {
+                using var _ = ArrayBuilder<IMethodSymbol>.GetInstance(out var linqMethodSymbolsBuilder);
+                foreach (var methodName in _nonEnumerableReturningLinqMethodNames)
+                {
+                    linqMethodSymbolsBuilder.AddRange(enumerableType.GetMembers(methodName).OfType<IMethodSymbol>());
+                }
+
+                if (linqMethodSymbolsBuilder.Count == 0)
+                {
+                    linqMethods = default;
+                    return false;
+                }
+
+                linqMethods = linqMethodSymbolsBuilder.ToImmutable() ;
+                return true;
+            }
+        }
+
+        public void AnalyzeInvocationOperation(OperationAnalysisContext context, ImmutableArray<IMethodSymbol> whereMethods, ImmutableArray<IMethodSymbol> linqMethods)
+        {
+            if (context.Operation.Syntax.GetDiagnostics().Any(diagnostic => diagnostic.Severity == DiagnosticSeverity.Error))
+            {
+                // Do not analyze linq methods that contain diagnostics.
+                return;
+            }
+
+            if (context.Operation is not IInvocationOperation invocation ||
+                invocation.Arguments.Length != 1)
+            {
+                return;
+            }
+
+            if (invocation.Arguments.Single().Children.FirstOrDefault() is not IInvocationOperation previousInvocationInChain)
```

Comment:
really don't like/want the FirstOrDefault on .Children.  we should just extract out the values we need in a strongly typed fashion that matches the exact shapes we expect.